dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,281 @@
+            }
+
+            uploadUrl = content.Uri.ToString();
+        }
+        else
+        {
+            throw new ArgumentException("AudioContent doesn't have any content.", nameof(content));
+        }
+
+        var transcriptId = await this.CreateTranscriptAsync(uploadUrl, executionSettings, cancellationToken)
+            .ConfigureAwait(false);
+        var transcript = await this.WaitForTranscriptToProcessAsync(transcriptId, executionSettings, cancellationToken)
+            .ConfigureAwait(false);
+
+        return new TextContent(
+            text: transcript.RootElement.GetProperty("text").GetString(),
+            modelId: null,
+            // TODO: change to typed object when AAI SDK is shipped
+            innerContent: transcript,
+            encoding: Encoding.UTF8,
+            metadata: null
+        );
+    }
+
+    /// <summary>
+    /// Transcribe audio file.
+    /// </summary>
+    /// <param name="content">Stream of the audio file</param>
+    /// <param name="executionSettings">The AI execution settings (optional).</param>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Text content from audio content.</returns>
+    public async Task<TextContent> GetTextContentAsync(
+        AudioStreamContent content,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(content);
+
+        string uploadUrl = await this.UploadFileAsync(content.Stream, cancellationToken).ConfigureAwait(false);
+
+        var transcriptId = await this.CreateTranscriptAsync(uploadUrl, executionSettings, cancellationToken)
+            .ConfigureAwait(false);
+        var transcript = await this.WaitForTranscriptToProcessAsync(transcriptId, executionSettings, cancellationToken)
+            .ConfigureAwait(false);
+
+        return new TextContent(
+            text: transcript.RootElement.GetProperty("text").GetString(),
+            modelId: null,
+            // TODO: change to typed object when AAI SDK is shipped
+            innerContent: transcript,
+            encoding: Encoding.UTF8,
+            metadata: null
+        );
+    }
+
+    private async Task<string> UploadFileAsync(Stream audioStream, CancellationToken ct)
+    {
+        var url = this.Url("v2/upload");
+        using var content = new StreamContent(audioStream);
+        content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
+        using var request = new HttpRequestMessage(HttpMethod.Post, url);
```

Comment:
Please separate logical blocks - getting the URL, creating the stream, creating the request, etc - with empty lines for better readability.